webauthn/helpers.js

```diff
@@ -537,28 +537,22 @@ function validateAuthenticatorAssertionResponse(assert) {
 
 function standardSetup(cb) {
     // Setup an automated testing environment if available.
-    let authenticator;
-    promise_test(async t => {
-        try {
-            authenticator = await window.test_driver.add_virtual_authenticator({
-                protocol: "ctap1/u2f",
-                transport: "usb"
-            });
-        } catch (error) {
-            if (error !== "error: Action add_virtual_authenticator not implemented") {
-                throw error;
-            }
-            // The protocol is not available. Continue manually.
-        }
-    }, "Set up the test environment");
-
-    cb();
-
-    promise_test(t => {
-        if (authenticator) {
-            return window.test_driver.remove_virtual_authenticator(authenticator);
+    window.test_driver.add_virtual_authenticator({
+        protocol: "ctap1/u2f",
+        transport: "usb"
+    }).then(authenticator => {
+        cb();
+        // XXX add a subtest to clean up the virtual authenticator since
+        // testharness does not support waiting for promises on cleanup.
+        promise_test(() => window.test_driver.remove_virtual_authenticator(authenticator),
+                     "Clean up the test environment");
```

Comment:
Did you mean
```suggestion
                     "Set up the test environment");
```

Reply:
No, the set-up happens on L540.

Reply:
Aha I see. `standardSetup` wraps all the tests in `cb`, which wasn't clear to me.